htlcswitch/switch.go

```diff
@@ -1627,15 +1645,16 @@ out:
 		// collect all the forwarding events since the last internal,
 		// and write them out to our log.
 		case <-s.cfg.FwdEventTicker.Ticks():
-			s.wg.Add(1)
-			go func() {
-				defer s.wg.Done()
-
-				if err := s.FlushForwardingEvents(); err != nil {
+			// The error of Go is ignored: if it is shutting down,
+			// the loop will terminate on the next iteration, in
+			// s.gm.Done case.
```

Comment:
another point for the above case :) 

This comment shows that the caller of `Go` should know that any error returned is just a shutting down error. I dont think the caller should be expected to know that.